sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -376,49 +381,57 @@ public class ChatClient {
                 if let error = error {
-                    completionHandler(
-                        .failure(AzureError.client("Failed to start push notifications", error)),
-                        response
-                    )
+                    completionHandler(.failure(AzureError.client("Failed to start push notifications", error)))
                 } else {
                     self.pushNotificationsStarted = true
-                    completionHandler(.success(()), response)
+                    completionHandler(.success(response))
                 }
             }
         } catch {
             options.logger.error("Failed to start push notifications with error: \(error)")
+            completionHandler(.failure(AzureError.client("Invalid URL.", error)))
         }
     }
 
+
     /// Stop push notifications.
     /// - Parameter completionHandler: Success indicates push notifications have been stopped.
     public func stopPushNotifications(
-        completionHandler: @escaping (Result<Void, AzureError>, HTTPURLResponse?) -> Void
+        completionHandler: @escaping (Result<HTTPURLResponse?, AzureError>) -> Void
     ) {
         guard pushNotificationsStarted else {
-            completionHandler(.failure(AzureError.client("Push notifications are not enabled.")), nil)
+            completionHandler(.failure(AzureError.client("Push notifications are not enabled.")))
+            return
+        }
+        
+        guard let registrarClient = registrarClient else {
+            completionHandler(.failure(AzureError.client("Failed to stop push notifications.")))
             return
         }
 
-        registrarClient?.deleteRegistration { response, error in
+        registrarClient.deleteRegistration { response, error in
             if let error = error {
-                completionHandler(.failure(AzureError.client("Failed to stop push notifications", error)), response)
+                completionHandler(.failure(AzureError.client("Failed to stop push notifications", error)))
             } else {
                 self.pushNotificationsStarted = false
-                completionHandler(.success(()), response)
+                completionHandler(.success(response))
             }
         }
     }
 
+    /// Handle the data payload for an incoming push notification.
+    /// - Parameters:
+    ///   - notification: The APNS push notification.
+    ///   - completionHandler: Receives the TrouterEvent from the push notification payload.
     /// Handle the data payload for an incoming push notification.
     /// - Parameters:
     ///   - notification: The APNS push notification.
     ///   - completionHandler: Receives the TrouterEvent from the push notification payload.
     public func handlePush(
         notification: [AnyHashable : Any],
-        completionHandler: TrouterEventHandler
+        completionHandler: PushNotificationEventHandler
```

Comment:
@JoshuaLai  given we have not GAd this sdk it is probably ok to make this change and add it to the release notes as a breaking change from the beta. We should be ok, there are not many users of the beta sdk (because it is beta still).

Reply:
Yes, you can make this change without issue. Just call it out in the changelog.